src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs

```diff
@@ -131,10 +131,69 @@ public PackageArchiveReader BuildPackage(PackageBuilder builder, string outputPa
                 PrintVerbose(outputPath, builder);
             }
 
+            if (_packArgs.OutputResolvedNuSpec)
+            {
+                WriteResolvedNuSpecToPackageOutputDirectory(outputPath, builder);
+            }
+
+            if (_packArgs.OutputSHA512PackageHash)
+            {
+                WriteSHA512PackageHash(outputPath, builder);
+            }
+
             _packArgs.Logger.LogMinimal(String.Format(CultureInfo.CurrentCulture, Strings.Log_PackageCommandSuccess, outputPath));
             return new PackageArchiveReader(outputPath);
         }
 
+        /// <summary>
+        /// Writes the resolved NuSpec file to the package output directory.
+        /// </summary>
+        /// <param name="packageOutputPath">The output package path</param>
+        /// <param name="packageBuilder">The package builder</param>
+        private void WriteResolvedNuSpecToPackageOutputDirectory(string packageOutputPath, PackageBuilder packageBuilder)
+        {
+            string packageOutputDirectory = Path.GetDirectoryName(packageOutputPath);
```

Comment:
we are doing something similar here:
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/MSBuildProjectFactory.cs#L90-L108

i'd rather have a similar implementation at both places. you just need to pass null for the FileCollection to Manifest constructor

Reply:
Nice, thank you for link. Updating to use the same implementation in the linked code.

Reply:
Resolved